src/item.cpp

```diff
@@ -3965,6 +3984,19 @@ bool item::ammo_consume( int qty ) {
     return false;
 }
 
+itype_id item::ammo_current() const
+{
+    if( is_ammo() ) {
+        return typeId();
```

Comment:
Basically, once you implemented `NULL_ID` like this (see trap.cpp for an actual example):
```C++
template<>
const string_id<Foo> string_id<Foo>::NULL_ID( "foo_null_id" );
```

You can now use `string_id<Foo>::is_null` and even the explicit `operator bool`:

```C++
string_id<Foo> id = ...;
if( id ) {
    // id is not null
} else {
    // id is equal to the null-id
}

// Or for function returning such:
string_id<Foo> get_foo_id() {
    if( some_condition ) {
        return valid_foo_id_from_somewhere();
    } else {
        return string_id<Foo>::NULL_ID;
    }
}
```

The only place you may need to document the actual value of the null id is on the JSON documentation when it's actually read from there.